ddtrace/internal/_encoding.pyx

```diff
@@ -794,7 +796,8 @@ cdef class MsgpackEncoderV05(MsgpackEncoderBase):
         if ret != 0:
             return ret
 
-        ret = msgpack_pack_map(&self.pk, len(span._meta) + (dd_origin is not NULL))
+        span_links = span._links_to_json()
```

Comment:
It might help to define the type of `span_links` here. I _think_ Cython will just have as a PyObject, but if we type as a string it can avoid some type checking later (iirc... checking the annotation would actually tell you).

Not expecting you to make this change, just calling out something fun to possibly look at.